src/lib.rs

```diff
@@ -258,6 +284,74 @@ The `Ready` state contains quite a bit of information, and you need to check and
 
 For more information, check out an [example](examples/single_mem_node/main.rs#L113-L179).
 
+## Membership Changes
+
+When building a resilient, scalable distributed system there is a strong need to be able to change
+the membership of a peer group *dynamically, without downtime.* This Raft crate supports this via
+**Joint Consensus**
+([Raft paper, section 6](https://web.stanford.edu/~ouster/cgi-bin/papers/raft-atc14)).
+
+It permits resilient arbitrary dynamic membership changes. A membership change can do any or all of
+the following:
+
+* Add peer (learner or voter) *n* to the group.
+* Remove peer *n* from the group.
+* Remove a leader (unmanaged, via stepdown)
+* Promote a learner to a voter.
+
+It (currently) does not:
+
+* Allow control of the replacement leader during a stepdown.
+* Optionally roll back a change during a peer group pause where the new peer group configuration
+fails.
```

Comment:
Does this mean it leaves the cluster in a failure state?

Reply:
No, the cluster would just be paused until it regains a majority.